web/modules/custom/config_form/src/Form/ConfigForm.php

```diff
@@ -0,0 +1,251 @@
+    ];
+    $form['action']['submit'] = [
+      '#type'      => 'submit',
+      '#value'     => $this->t('submit'),
+      '#ajax'      => [
+        'callback' => '::submitDataAjax',
+      ],
+    ];
+    $form['success'] = [
+      '#type'     => 'markup',
+      '#markup'   => '<div class="success"></div>',
+    ];
+    return $form;
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public function validateForm(array &$form, FormFormStateInterface $form_state) {
+    $this->messenger()->addMessage($this->validate($form_state));
+  }
+
+  /**
+   * This function is used to facilitate form submission without page refresh .
+   *
+   * @param array $form
+   *   This array contains all the form field in an associative array format.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   This array holds the current state of the form with input data.
+   *
+   * @return \Drupal\Core\Ajax\AjaxResponse
+   *   Ajax response based on the validation.
+   */
+  public function submitDataAjax(array &$form, FormFormStateInterface $form_state) {
+    $ajax_response = new AjaxResponse();
+    $output = $this->validate($form_state);
+    if ($output) {
+      // If validation is succesful showing the success message.
+      $message = $this->t('Thanks for submitting the form');
+      $ajax_response->addCommand(new CssCommand('.success', ['color' => 'green']));
+      $ajax_response->addCommand(new HtmlCommand('.success', $message));
+      $ajax_response->addCommand(new CssCommand('#error-message', ['display' => 'none']));
+    }
+    else {
+      // If validation failed showing the error message.
+      $ajax_response->addCommand(new CssCommand('#error-message', ['color' => 'red']));
+      $ajax_response->addCommand(new HtmlCommand('#error-message', $output));
+    }
+    // Deleting all messages after process is completed.
+    $this->messenger()->deleteAll();
+    return $ajax_response;
+  }
+
+  /**
+   * This function is used to facilitate the validation of data entered by user.
+   *
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   Holds the form state along with the input data.
+   *
+   * @return mixed
+   *   On succesful validation return boolean value
+   *   else return error message.
+   */
+  public function validate(FormFormStateInterface $form_state) {
+    $phone_number = $form_state->getValue('phone_number');
+    $email = $form_state->getValue('email');
+    $config = $this->config('config_form.settings');
+    $config_values = $config->get();
+    $allowed_domains = $config_values['allowed_domains'];
+    // Exploding the array so now we have the data in this format
+    // if email = 'example.com'
+    // Array ( [0] => example [1] => gmail.com )
+    $parts = explode('@', $email);
+    // Pop function removes the last element from an array.
+    $domain = array_pop($parts);
+    // Validating based on conditions.
+    if (!filter_var($email, FILTER_VALIDATE_EMAIL)) {
```

Comment:
Can we use switch case?

Reply:
We can , but how would that make the code efficient? Because the time complexity will still be the same that is O(n)